### DISCUSSION TITLE
How to create multi panel with roles

### DISCUSSION THREAD

**Discussion by Youssef-Elouasbi** [filamentphp/filament#13781] [Help]
### Package

Panel builder

### Package Version

3.2.61

### How can we help you?

Hi
how can i add multi panel based on user role
for example i want to have two panels if user role is "sport" i want him to see panel "sport" if not i want him to see default panel in filament php project after login

**Comment by dissto** [comment]
Did you check out [Authorizing access to the panel](https://filamentphp.com/docs/3.x/panels/users#authorizing-access-to-the-panel) ? 🤔

**Reply by Youssef-Elouasbi** [reply]
yes i have this function in my user model
 public function canAccessPanel(Panel $panel): bool
    {
        return true;
    }
    
    but what i should put inside of this function should redirect user to a specific panel after login !? but each panel has different login page
    
please can you give me step by step i am really confused

**Reply by Youssef-Elouasbi** [reply]
should i do like this for my case !?

    public function canAccessPanel(Panel $panel): bool
    {
        if ($panel->getId() === 'sport' && !$this->hasRole('sport')) {
            return false;
        } 
        return true;
    }

**Reply by dissto** [reply]
Come to think of it, you probably need a middleware and redirect the user based on their role to the correct panel 🤔 The docs do not have an example (at least I couldn't find it).

Here is an [example](https://laraveldaily.com/post/filament-disable-dashboard-auto-redirect-url-after-login) (but applied to the login). Maybe you can extract the bits you need?

See also: [Applying middleware](https://filamentphp.com/docs/3.x/panels/configuration#applying-middleware)

See also: https://github.com/filamentphp/filament/discussions/12238#discussioncomment-9040580
